File: src/main/java/day03/DataFrameOperator.java
package day03;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.RowFactory;
import org.apache.spark.sql.SQLContext;
import org.apache.spark.sql.types.DataTypes;
import org.apache.spark.sql.types.StructField;
import org.apache.spark.sql.types.StructType;

import java.util.ArrayList;

import static org.apache.spark.sql.types.DataTypes.IntegerType;
import static org.apache.spark.sql.types.DataTypes.StringType;


public class DataFrameOperator {
    public static void main(String[] args){

        SparkConf conf = new SparkConf().setAppName("DataFrameOperator").setMaster("local");

        JavaSparkContext sc = new JavaSparkContext(conf);

        SQLContext sqlContext = new SQLContext(sc);

        JavaRDD<String> line = sc.textFile("people.txt");
        //构建rowRDD
        JavaRDD<Row> rowRDD = line.map(new Function<String, Row>() {
            private static final long serialVersionUID = 1L;

            public Row call(String line) throws Exception {
                String[] split = line.split(",");
                return RowFactory.create(Integer.valueOf(split[0]), split[1], Integer.valueOf(split[2]));
            }
        });
        //构建schme
        ArrayList<StructField> structFields = new ArrayList<StructField>();

        structFields.add(DataTypes.createStructField("id",IntegerType,true));

        structFields.add(DataTypes.createStructField("name",StringType,true));

        structFields.add(DataTypes.createStructField("age",IntegerType,true));

        StructType structType = DataTypes.createStructType(structFields);

        Dataset<Row> dataFrame = sqlContext.createDataFrame(rowRDD, structType);

        dataFrame.registerTempTable("people");

        Dataset<Row> row = sqlContext.sql("select * from people where age>10");

        row.show();

        sc.stop();
    }
}

File: src/main/scala/day17/DataFrameOperator.scala
package day17

import java.util
import java.util.ArrayList

import org.apache.spark.{SparkConf, SparkContext}
import org.apache.spark.sql._
import org.apache.spark.sql.types._

object DataFrameOperator {
  def main(args: Array[String]): Unit = {
    val conf: SparkConf = new SparkConf().setAppName("DataFrameOperator").setMaster("local")

    val sc: SparkContext = new SparkContext(conf)

    val session = SparkSession.builder().appName("DataFrameOperator").master("local").getOrCreate()

    val line = sc.textFile("people.txt")
    //构建lineRDD
    val lineRDD = line.map(_.split(","))
    //构建rowRDD
    val rowRDD = lineRDD.map(x=>Row(x(0).toInt,x(1),x(2).toInt))
    //构建schme
    val schma = StructType(
      List(
        StructField("id",IntegerType,true),
        StructField("name",StringType,true),
        StructField("age",IntegerType,true)
      )
    )
    //构建df
    val df = session.createDataFrame(rowRDD,schma)
    //创建临时表
    df.createTempView("people")
    //执行sql
    val row: Dataset[Row] = session.sql("select * from people where age>10")
    //
    row.show()

    sc.stop()
  }
}
